packages/core/test/invoke.test.ts

```diff
@@ -2050,6 +2053,116 @@ describe('invoke', () => {
     });
   });
 
+  describe('with behaviors', () => {
+    it('should work with a behavior', (done) => {
+      const countBehavior: Behavior<EventObject, number> = {
+        receive: (count, event) => {
+          if (event.type === 'INC') {
+            return count + 1;
+          } else {
+            return count - 1;
+          }
+        },
+        initial: 0
+      };
+
+      const countMachine = createMachine({
+        invoke: {
+          id: 'count',
+          src: () => countBehavior
+        },
+        on: {
+          INC: {
+            actions: forwardTo('count')
+          }
+        }
+      });
+
+      const countService = interpret(countMachine)
+        .onTransition((state) => {
+          if (state.children['count']?.getSnapshot() === 2) {
+            done();
+          }
+        })
+        .start();
+
+      countService.send('INC');
+      countService.send('INC');
+    });
+
+    it('behaviors should have reference to the parent', (done) => {
+      const pongBehavior: Behavior<EventObject, undefined> = {
+        receive: (_, event, { parent }) => {
+          if (event.type === 'PING') {
+            parent?.send({ type: 'PONG' });
+          }
+
+          return undefined;
+        },
+        initial: undefined
+      };
+
+      const pingMachine = createMachine({
+        initial: 'waiting',
+        states: {
+          waiting: {
+            entry: send('PING', { to: 'ponger' }),
+            invoke: {
+              id: 'ponger',
+              src: () => pongBehavior
```

Comment:
Do behaviours passed into invoke.src always need to be like this? Or can they be declared like this:

`src: pongBehavior`

Reply:
That's a good question - sometimes, behaviors need to be dynamically instantiated with data from context (i.e., actors can be "born with innate knowledge"). So then the question becomes whether we want to support `src: () => behavior` as well as `src: behavior`, or avoid confusion and say that the signature must be:

```js
src: (context, event) => behavior
```

This is how it is in V5, btw ☝️